### DISCUSSION TITLE
Suggest Custom Directives to listen for changes in 'v-show' switching.

### DISCUSSION THREAD

**Discussion by coderyqy** [vuejs/rfcs#650] [General]
There are three components on the page, which are displayed and hidden using the v-show control. Each component uses my custom command "v-aniFromLeft". My expectation is that the current component will have a callback when displayed or hidden, to facilitate subsequent operations.

I know the updated lifecycle function,
```js
  // called after the parent component and
  // all of its children have updated
  updated(el, binding, vnode, prevVnode) {},
```
 which can listen for changes, but when other components switch, updated will also execute, which can result in many unnecessary calls.

Thank you for taking the time to read this question. If there is a better way, we can also discuss it.

**Comment by posva** [comment]
Please, use the [Discord chat](https://chat.vuejs.org/) to ask questions
